src/modules/ekf2/EKF/ekf_helper.cpp

```diff
@@ -432,6 +433,24 @@ bool Ekf::setEkfGlobalOrigin(const double latitude, const double longitude, cons
 		_pos_ref.initReference(latitude, longitude, _time_delayed_us);
 		_gps_alt_ref = altitude;
 
+		const float mag_declination_gps = get_mag_declination_radians(latitude, longitude);
```

Comment:
In a way this feels like a potential hole if someone sets it questionably, but I suppose if we don't have real GPS there's nothing we can do about it.